src/lightning/pytorch/core/hooks.py

```diff
@@ -151,6 +152,13 @@ def on_predict_batch_end(self, outputs: Optional[Any], batch: Any, batch_idx: in
 
         """
 
+    def on_validation_model_zero_grad(self) -> None:
+        """Called by the training loop to release gradients before entering the validation loop."""
+        if _TORCH_GREATER_EQUAL_2_0:
+            self.zero_grad()
+        else:
+            self.zero_grad(set_to_none=True)
```

Comment:
That wouldn't be the same as `model.zero_grad()`. The loop never called optimizer's `zero_grad`. The motivation of the hook is to provide the override for specifically only the call that happens at the transition from training to validation. If you did an override of `LightningModule.zero_grad()`, you'd have to add logic to check whether you're in validation.

Reply:
Oh sorry, I mixed the two methods.

>  If you did an override of LightningModule.zero_grad(), you'd have to add logic to check whether you're in validation.

This is the point I was trying to make. Wouldn't it be fine to suggest this? Checking whether you're in validation is simple since `self` has a trainer reference

Reply:
I wanted to bring this in line with the other hooks we have for `model.eval()` at the training-validation boundary. This way, the user wouldn't override `nn.Module's zero_grad()` that applies more generally to the model rather than the training loop. A clear separation between Trainer hook and model method would be desireable I think.

Reply:
But `zero_grad` would be overridden in the `LightningModule`, not the underlying `nn.Module`. The LM already defines how the trainer interacts with the underlying module so from my point of view, it fits well.

It's fine if you still want the new hook, I just thought that we could avoid having one more hook to remember about

Reply:
I'd like to keep it unless there are stronger arguments against it.

I see two clear benefits in doing it this way. 1) The `zero_grad()` method remains untouched and can be used on the LightningModule outside the trainer context 2) Overriding the hook is cleaner because the alternative would mean the user would have to None-check `self.trainer` and check the trainer stage for validation and then also shouldn't forget to all super().zero_grad().